pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -383,24 +384,18 @@ func (s *Reconciler) Exists(ctx context.Context) (bool, error) {
 		}
 	}
 
-	// VM States come from https://docs.microsoft.com/en-us/azure/virtual-machines/windows/states-lifecycle#provisioning-states
-	switch v1beta1.VMState(*vm.ProvisioningState) {
-	case v1beta1.VMStateSucceeded:
-		// VM was created or updated successfully
-		klog.Infof("Machine %v is running", to.String(vm.VMID))
-	case v1beta1.VMStateUpdating:
-		// Some update is being applied
-		klog.Infof("Machine %v is updating", to.String(vm.VMID))
-	case v1beta1.VMStateCreating:
-		// Creation request was accepted, VM is being created
-		klog.Infof("Machine %v is creating", to.String(vm.VMID))
-	default:
-		klog.Infof("Not found vm for machine %s", s.scope.Machine.GetName())
+	// VM provisioning states description https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/virtual-machines/states-billing.md
+	// However I didn't found any documentation about concrete ProvisioningState values.
+	// VMState type looks correct, for only Failed vms treat as non existed here.
+	// Actually deleted vms would be catch by logic above.
+	vmProvisioningState := v1beta1.VMState(*vm.ProvisioningState)
+
+	if vmProvisioningState == v1beta1.VMStateFailed {
```

Comment:
Looking at the list, what about deallocating/deallocate? Does this need to be accounted for? I think this is what happens when the VM is being deleted outside of the scope of Machine API, in which case I think we want to go failed in this case, so I think these should also be included
```suggestion
	switch vmProvisioningState {
	case v1beta1.VMStateFailed, v1beta1.VMStateDeallocating, v1beta1.VMStateDeallocated:
```

Could use a switch and then interpolate the actual state into the log message?

https://github.com/openshift/cluster-api-provider-azure/blob/b024905873bbc955a432559a9af923a6f002a4ae/pkg/apis/azureprovider/v1beta1/types.go#L342-L345

Reply:
I don't think that we need to account this here. This list is a mix between ProvisioningState and PowerState of vm. In case of vm deletion ProvisioningState would indicate `Deleting` for some period of time.

However `Deallocating/Deallocated` happens when vm was just powered off but not deleted. Not sure that we should treat it as non existing.

Here is a bit more context: https://github.com/openshift/cluster-api-provider-azure/pull/219/files#diff-af805e1e45f03df0b5b56ff4413e5ad52cd31904a94d37e8e916751953e4687dR295